lib/reek/smell_detectors/base_detector.rb

```diff
@@ -122,6 +124,25 @@ def valid_detector?(detector)
           descendants.map { |descendant| descendant.to_s.split('::').last }.
             include?(detector)
         end
+
+        #
+        # @param detector_name [String] the detector in question, e.g. 'DuplicateMethodCall'
+        # @return [SmellDetector] - this will return the class, not an instance
+        #
+        def to_detector(detector_name)
+          SmellDetectors.const_get detector_name
+        end
+
+        #
+        # @return [Set<Symbol>] - all configuration keys that are available for this detector
+        #
+        def configuration_keys
+          BASIC_OPTIONS + custom_options
```

Comment:
Sorry, but I don't follow. Where's `default_options` coming from, where's the benefit of doing it like that and how is `Set.new(default_options.keys.map(&:to_sym)` related to `BASIC_OPTIONS + custom_options`?